pkg/jmxfetch/jmxfetch.go

```diff
@@ -226,14 +227,35 @@ func (j *JMXFetch) Start() error {
 	return j.cmd.Start()
 }
 
-// Kill kills the JMXFetch process
-func (j *JMXFetch) Kill() error {
+// Stop stops the JMXFetch process
+func (j *JMXFetch) Stop() error {
 	if j.JmxExitFile == "" {
+		graceTimer := time.NewTimer(500 * time.Millisecond)
+		defer graceTimer.Stop()
+
+		stopped := make(chan struct{})
+
 		// Unix
-		err := j.cmd.Process.Signal(os.Kill)
+		err := j.cmd.Process.Signal(syscall.SIGTERM)
 		if err != nil {
 			return err
 		}
+
+		go func() {
+			j.Wait()
+			close(stopped)
+		}()
+
+		select {
+		case <-graceTimer.C:
+			log.Warnf("Jmxfetch did not exit during it's grace period, killing it")
+			err = j.cmd.Process.Signal(os.Kill)
```

Comment:
There's an ongoing race-condition here if the process doesn't come down in 500ms but is shutting down and does so before this code is reached. This would then return an error, it's no biggie as I think all we'd do is log the error, but let's make sure the log-line is meaningful.